ggml-metal.m

```diff
@@ -12,18 +12,16 @@
 #else
 #define metal_printf(...) fprintf(stderr, __VA_ARGS__)
 #endif
+#define metal_printf(...) fprintf(stderr, __VA_ARGS__)
 
 #define UNUSED(x) (void)(x)
 
-struct ggml_metal_buffer {
-    const char * name;
-
-    void   * data;
-    size_t   size;
-
-    id<MTLBuffer> metal;
+struct ggml_metal_buffer_wrapper {
+    id<MTLBuffer> buffer;
 };
 
+static void * g_ptr_base = (void *)0x1234;
```

Comment:
This should be a multiple of the alignment, otherwise you will lose the first few bytes of the buffer, or worse, the tensors may end not being aligned. Something like `0x1000` should work.